Regression mvnd 1.0-m6 vs. mvn 3.9.1

```bash
mvnd --version
Apache Maven Daemon (mvnd) 1.0-m6 darwin-aarch64 native client (47d4c6b9c399079d91312bb8ae35d1d76e7fd97a)
Terminal: org.jline.terminal.impl.PosixSysTerminal with pty org.jline.terminal.impl.jansi.osx.OsXNativePty
Apache Maven 3.9.1 (2e178502fcdbffc201671fb2537d0cb4b4cc58f8)
Maven home: /opt/homebrew/Cellar/mvnd/1.0-m6-m39/libexec/mvn
Java version: 17.0.6, vendor: Eclipse Adoptium, runtime: /Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "13.3.1", arch: "aarch64", family: "mac"
```

```bash
mvn --version
Apache Maven 3.9.1 (2e178502fcdbffc201671fb2537d0cb4b4cc58f8)
Maven home: /opt/homebrew/Cellar/maven/3.9.1/libexec
Java version: 17.0.6, vendor: Eclipse Adoptium, runtime: /Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "13.3.1", arch: "aarch64", family: "mac"
```

how to reproduce:

```bash
% git clone git@github.com:jdbi/jdbi
% cd jdbi
% git checkout v3.38.0-rc3
% mvnd -pl :jdbi3-core -am clean install
[...]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for jdbi3 build parent 3.38.0-rc3:
[INFO]
[INFO] jdbi3 - internal - policy .......................... SUCCESS [  1.670 s]
[INFO] jdbi3 - internal - build parent .................... FAILURE [  0.095 s]
[INFO] jdbi3 build parent ................................. SKIPPED
[INFO] jdbi3 core ......................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.828 s (Wall Clock)
[INFO] Finished at: 2023-04-24T20:23:59-07:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.2.1:enforce (enforce-jdbi) on project jdbi3-build-parent:
[ERROR] Rule 1: org.apache.maven.enforcer.rules.property.RequireProperty failed with message:
[ERROR] This module must set a moduleName.
[ERROR] -> [Help 1]
```

Same command with mvn works fine.

Given the way our project sets module names (we use per-module profile activation based on the presence and absence of files in the directories), the most likely culprit is some sort of caching of profile activation.  The correct list of profile activations would be:

```
Active Profiles for Project 'org.jdbi.internal:jdbi3-policy:jar:3.38.0-rc3':

The following profiles are active:




Active Profiles for Project 'org.jdbi.internal:jdbi3-build-parent:pom:3.38.0-rc3':

The following profiles are active:

 - in-parent (source: org.jdbi.internal:jdbi3-build-parent:3.38.0-rc3)
 - macos arm (source: org.jdbi.internal:jdbi3-build-parent:3.38.0-rc3)



Active Profiles for Project 'org.jdbi:jdbi3-parent:pom:3.38.0-rc3':

The following profiles are active:

 - java17 (source: org.jdbi:jdbi3-parent:3.38.0-rc3)
 - in-parent (source: org.jdbi.internal:jdbi3-build-parent:3.38.0-rc3)
 - macos arm (source: org.jdbi.internal:jdbi3-build-parent:3.38.0-rc3)



Active Profiles for Project 'org.jdbi:jdbi3-core:jar:3.38.0-rc3':

The following profiles are active:

 - java17 (source: org.jdbi:jdbi3-parent:3.38.0-rc3)
 - macos arm (source: org.jdbi.internal:jdbi3-build-parent:3.38.0-rc3)
 - basepom.invoker-reporting (source: org.basepom:basepom-oss:52)
 - basepom.invoker-integration-testing (source: org.basepom:basepom-foundation:52)
```

The `in-parent` profile provides a module name. Running the same command with `mvnd` yields no output (!).

output of `mvnd -X -pl :jdbi3-core -am clean install`

[logfile-mvnd.txt](https://github.com/apache/maven-mvnd/files/11316162/logfile-mvnd.txt)

output of `mvn -X -pl :jdbi3-core -am clean install`

[logfile-mvn.txt](https://github.com/apache/maven-mvnd/files/11316160/logfile-mvn.txt)

--- a/daemon/pom.xml
+++ b/daemon/pom.xml
@@ -61,7 +61,6 @@
     <dependency>
       <groupId>io.takari.maven</groupId>
       <artifactId>takari-smart-builder</artifactId>
-      <version>0.6.2</version>
     </dependency>
 
     <!-- Logging -->

--- a/pom.xml
+++ b/pom.xml
@@ -116,7 +116,7 @@
     <maven-assembly-plugin.version>3.3.0</maven-assembly-plugin.version>
     <xstream.version>1.4.20</xstream.version>
     <plexus-interactivity-api.version>1.0</plexus-interactivity-api.version>
-    <takari-smart-builder.version>0.6.2</takari-smart-builder.version>
+    <takari-smart-builder.version>0.6.3-SNAPSHOT</takari-smart-builder.version>
   </properties>
 
   <dependencyManagement>

--- a/pom.xml
+++ b/pom.xml
@@ -116,7 +116,7 @@
     <maven-assembly-plugin.version>3.3.0</maven-assembly-plugin.version>
     <xstream.version>1.4.20</xstream.version>
     <plexus-interactivity-api.version>1.0</plexus-interactivity-api.version>
-    <takari-smart-builder.version>0.6.3-SNAPSHOT</takari-smart-builder.version>
+    <takari-smart-builder.version>0.6.3</takari-smart-builder.version>
   </properties>
 
   <dependencyManagement>

